repo_health/repo_health.py

```diff
@@ -0,0 +1,127 @@
+"""
+pytest plugin to test if specific repo follows standards
+"""
+import os
+from collections import defaultdict
+
+import pytest
+
+import yaml
+
+session_data_holder_dict = defaultdict(dict)
+
+@pytest.fixture(scope="session")
+def all_results():
+    return session_data_holder_dict
+
+def pytest_configure(config):
+    """
+    pytest hook used to add plugin install dir as place for pytest to find tests
+    """
+
+    if config.getoption("repo_health"):
+
+        # Add path to pytest-repo-health dir so pytest knows where to look for checks
+        file_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+        config.args.append(file_dir)
+
+        # add repo path so a repo can design their own checks inside a repo_state_checks dir
+        repo_path = config.getoption("repo_path")
+        if repo_path is None:
+            repo_path = os.getcwd()
+        config.args.append(os.path.abspath(repo_path))
+
+        # in case repo_health checks are in seperate repo
+        repo_health_path = config.getoption("repo_health_path")
```

Comment:
We'll want to switch to a plugin mechanism probably based on importlib-metadata pretty soon, but this can work for the first iteration.